doc/wrapping_js_components.md

```diff
@@ -0,0 +1,699 @@
+
+```dart
+@JS('MagicalJsPackage.ArbitraryComponent')
+```
+
+and would correspond to `window.MagicalJsPackage.ArbitraryComponent`.
+
+> Reminder: The JavaScript bundle for the JS must be loaded into the application, [potentially similar to how React is added][react-script], or else this interop step will always fail at runtime.
+
+To check that the name is correct, you can use the browser devtools to attempt to access the property on the `window`. If the property is valid and generally where you expect it to be then it's likely correct.
+
+<img src='./images/ui_js_component/interop_ref.gif' alt='Accessing the property in the devtools' />
+
+If the annotation does not have the correct name, rendering the component will fail with a runtime error that looks like:
+
+<img src='./images/ui_js_component/interop_reference_error.png' alt='Accessing the property in the devtools' />
+
+> **One more reminder** that if after correctly creating the annotation, you find that it does not have a library prefix and only requires the component name (i.e. `JS('ArbitraryComponent')` and not something like `JS('MagicalJsPackage.ArbitraryComponent')`), you should **heavily** consider configuring your bundle to have at least a package namespace (i.e. the 'MagicalJsPackage' prefix). Loading components to the global namespace both pollutes that namespace as well as risks naming collisions.
+
+### Adding Prop Typings
+
+Next, you'll want to create the props mixin for your component. A props mixin declares props, types them, and in more advanced cases converts their values. In simple cases, you're just adding a property to the props mixin that lines up with a prop on the JS side, and [react-dart][react-dart] will handle converting that data from Dart to JavaScript. Meanwhile, other cases require manual type conversions to form the data in a way that JS can understand.
+
+This section will give a basic example of what writing this mixin looks like and then give additional examples of more complex type conversions.
+
+#### Props Conversion Example
+
+Start by adding your component imports to your file, if you haven't already.
+
+```dart
+import 'package:over_react/over_react.dart';
+
+part 'arbitrary_component.over_react.g.dart';
+```
+
+These are just the standard OverReact component directives. Altogether, the imports section of your file should look something like:
+
+```dart
+@JS()
+library some_library.path.to.component;
+
+import 'package:js/js.dart';
+import 'package:over_react/over_react.dart';
+
+part 'arbitrary_component.over_react.g.dart';
+```
+
+After that, move on to converting the TS props interface to a Dart mixin. Say you are given a props interface (or type) that looks like:
+
+```ts
+interface ArbitraryComponentProps {
+  anObjProp: Record<string, bool>;
+  content: string;
+  onClick: (e: React.MouseEvent, id: string) => void;
```

Comment:
```suggestion
  onClick: (e: React.SyntheticMouseEvent, id: string) => void;
```

Reply:
Also - this "intricate" conversion seems like something that may be unique to MUI JS... Should we mention that if the JS component just uses the HTML `onClick` function signature - that the namespacing is not necessary?

Reply:
Or maybe include a dom callback like `onBlur` as well - that doesn't require a custom conversion and be really explicit that the `onClick` used in the example is for demonstration purposes only - and that most JS component libs don't customize dom prop callbacks like this?

Reply:
In TS, is it `React.SyntheticMouseEvent`? I only saw [types like these](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/5a03f363621433711005972e084f8e2111662299/types/react/v17/index.d.ts#L1156-L1160), where the name is just something like `ClipboardEvent` and it extends from `SyntheticEvent` (or `UiEvent`, which extends from `SyntheticEvent`).

I agree though that we should specific this is a rare case. I'll update that!

Reply:
mmm.... you might be right that its not a synthetic event in dom callback cases in typescript land.

Reply:
Ohh I may be misunderstanding, but I don't think that's quite what I was saying. I meant that in TS, synthetic events don't have the word "synthetic" in them. A `React.MouseEvent` will map to a Dart `SyntheticMouseEvent`